tests/src/Unit/EventTrackerServiceTest.php

```diff
@@ -0,0 +1,362 @@
+<?php
+
+namespace Drupal\Tests\commerce_google_tag_manager\Unit;
+
+use Drupal\commerce_order\Entity\OrderInterface;
+use Drupal\commerce_order\Entity\OrderItemInterface;
+use Drupal\commerce_price\Price;
+use Drupal\commerce_product\Entity\ProductInterface;
+use Drupal\commerce_product\Entity\ProductVariationInterface;
+use Drupal\commerce_store\Entity\StoreInterface;
+use Drupal\Tests\UnitTestCase;
+use Drupal\commerce_google_tag_manager\EventTrackerService;
+use Drupal\commerce_google_tag_manager\EventStorageService;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+/**
+ * Tests for the EventTracker service.
+ *
+ * @coversDefaultClass \Drupal\commerce_google_tag_manager\EventTrackerService
+ *
+ * @group commerce
+ * @group commerce_google_tag_manager
+ * @group commerce_google_tag_manager_unit
+ */
+class EventTrackerServiceTest extends UnitTestCase {
+
+  /**
+   * The Commerce GTM event tracker.
+   *
+   * @var \Drupal\commerce_google_tag_manager\EventTrackerService
+   */
+  private $eventTracker;
+
+  /**
+   * The commerce GTM event storage.
+   *
+   * @var \Drupal\commerce_google_tag_manager\EventStorageService
+   */
+  private $eventStorage;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+
+    $this->eventStorage = $this->getMockBuilder(EventStorageService::class)
+      ->disableOriginalConstructor()
+      ->getMock();
+
+    $event_dispatcher = $this->prophesize(EventDispatcherInterface::class);
```

Comment:
Maybe we should use the `getMockBuilder`. I don't know if we should stick to 1 or mixed both.
When I contribute to the Core I always use `getMockBuilder` because it's used 80% of the time. But for me both are fine (I personally prefer Prophecy).